Move some of the deprecated tasks stuff outside of the main components

louwie17 opened the pull request:
Move some of the deprecated tasks stuff outside of the main components.
These are the client facing changes suggested in https://github.com/woocommerce/woocommerce-admin/pull/7749 (@joshuatf  hope I didn't miss anything here)

No changelog necessary

### Screenshots

### Detailed test instructions:

- Load this branch and install the Google Listings & Ads plugin ( available under the marketing tools task)
- Notice the new task list (Things to do next) click on `Set up Google Listings & ads`, it should correctly redirect to the plugins set up screen.
- Run `npm run example -- --ext=add-task` locally and enable the **WooCommerce Admin add task** plugin
- Click on the **Example** task displayed in the extended task list, it should render the container with the **Mark task complete** button.
- Add another task with a different `key` to [the example](https://github.com/woocommerce/woocommerce-admin/blob/main/docs/examples/extensions/add-task/js/index.js) with `level: 1` and make sure it shows up at the top of the list.
- Then update the Example task, setting `completed` to `true`, and make sure it's displayed at the bottom of the list now.
- The main task list should not be affected by this sorting.

louwie17 committed:
Move some of the deprecated tasks stuff outside of the main components

louwie17 committed:
Do not add data to request when no depreciated tasks

joshuatf reviewed (commented):
Awesome work!  Could we also move the `possiblyPruneTaskData` logic in here?  Looks like we might be able to use the `isDeprecated` flag to verify that data now.

louwie17 committed:
Move possiblePruneTaskData function to the deprecated-tasks class

louwie17 commented:
> Awesome work! Could we also move the `possiblyPruneTaskData` logic in here? Looks like we might be able to use the `isDeprecated` flag to verify that data now.

@joshuatf I moved the above function to the same class, as a static function, given we don't need to call the `applyFilters` again.
I am not able to use the `isDepreciated` in this case, as the task we pass in is coming from the API.

I also removed the E2E test changes and created a separate PR for that.

joshuatf reviewed (approved):
>I am not able to use the isDepreciated in this case, as the task we pass in is coming from the API.

Ah yes, my mistake.

>I also removed the E2E test changes and created a separate PR for that.

👍 

Looks great and is testing well; thank you very much for the clean-up!  :shipit:
